### PR TITLE
Storage and replication controllers proposal.

### PR DESCRIPTION
**Pull request by childsb**
This is a draft proposal for storage volume handling with replication controllers.

### CODE REVIEW

**Comment by bgrant0607** [comment]
See #18016.

cc @thockin @smarterclayton

File: docs/design/storage_replication.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+# Goals
+
+* Determine how a replication controller should map replicas to volumes
+* Determine how volumes are handled when an RC scales up or down.
+* Define the code design and markup changes required for Replication Controllers to assign unique volumes to each replica
+
+# Assigning Volumes to Replicas
+There are two potential volume type assignments for replication controllers- existing unclaimed volumes and non-existant dynamically provisioned volumes.
+
```
Comment (markturansky) [inline docs/design/storage_replication.md]: Also shared storage.  The current implementation only supports shared due to the single claimName in the PodTemplate, which is good for shared.  We'd need to preserve this functionality while also allowing a distinct volume per replica.

**Comment by smarterclayton** [comment]
I don't believe this should be solved with this proposal - everything described here more properly belongs in a pet set.

File: docs/design/storage_replication.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+## Dynamically provisioned Volumes
+A user would specify a provisioner label in the replicated pod definition and the replication controller would provision new volumes for each replica.  **Note:** The dynamic provisioning feature is still in alpha.  
+
+## Order of replicas claiming volumes
+Replicas will first get volumes from the list of available PVs then dynamically provision PVs as needed if a DP selector is supplied.
+
+## Scaling up and down
+* Scale up - When a replication controller is scaled up new PV are acquired first from the listed pool of predefined PVs.  Once the pool is empty (or if there is none) PVs are dynamically provisioned based on the provisioning selector.
+
```
Comment (markturansky) [inline docs/design/storage_replication.md]: Do you mean "predefined PVCs"?

**Comment by thockin** [comment]
Such docs really should start with use-cases and behavioral requirements.  "Why" is far more important than "how".

**Comment by fejta** [comment]
This PR has had no meaningful activity for multiple months. If it is still valid please rebase, push a new commit and reopen the PR. Thanks!

**Pull request closed without merging**